crates/locutus-core/src/ring.rs

```diff
@@ -529,6 +542,15 @@ impl Ord for Distance {
     }
 }
 
+impl Sub<Distance> for f64 {
+    type Output = Distance;
+
+    fn sub(self, rhs: Distance) -> Self::Output {
+        debug_assert!((0.0..=1.0).contains(&self));
+        Distance::new((rhs.0 - self).abs())
```

Comment:
Same issue, abs may violate expectations for - operator

Reply:
We are subtracting ring distances which cannot be negative (since there is no sense of direction afaik) one of the advantages of being able to impl the op on the type itself is so we can enforce the semantics of it at the type level.

The alternative would be to call abs on the result or implement abs for distance anyway, so I don't see much benefit to that.

Reply:
I was thinking to use an explicit distance function on Location rather than overloading the - operator to avoid the possibility of confusion, but it's not a big deal.